app/assets/javascripts/directives/hostname_validation.js

```diff
@@ -0,0 +1,33 @@
+(function() {
+  angular.module('ManageIQ').directive('hostnameValidation', directive);
+
+  function directive() {
+    return {
+      require: 'ngModel',
+      restrict: 'A',
+      link: link,
+    };
+    function link(_scope, _elem, _attrs, ctrl) {
+      ctrl.$validators.hostnameValidation = function(_modelValue, viewValue) {
+        return ! viewValue || validateHostname(viewValue);
+      };
+      /**
+       * The hostname is invalid if:
+       *  - Has protocol before the address (ex: http://);
+       *  - If it is an IP address and has some path after the address;
+       * @param {any} hostname - Hostname to be validate
+       * @returns {boolean} True if the hostname is valid
+       */
+      var validateHostname = function(hostname) {
+        return ! hasProtocol(hostname) && ! isIpAddressWithPath(hostname);
+      };
+      var hasProtocol = function(hostname) {
+        return /\b:\/\/\b/.test(hostname);
+      };
+      var isIpAddressWithPath = function(hostname) {
+        var isIpAddress = /\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(hostname);
+        return isIpAddress && hostname.indexOf('/') > 0;
+      };
+    }
+  }
+})();
```

Comment:
If we consider removing the path (see current discussion), here is my suggestion for this code (needs testing):

```javascript
(function() {
  angular.module('ManageIQ').directive('hostnameValidation', directive);

  function directive() {
    return {
      require: 'ngModel',
      restrict: 'A',
      link: link,
    };
    function link(_scope, _elem, _attrs, ctrl) {
      ctrl.$validators.hostnameValidation = function(_modelValue, viewValue) {
        return isHostname(viewValue) || isIpAddress(viewValue);
      };

      /**
       * Verifies if the informed value is a valid hostname.
       *
       * @param {string} value - Value to be validated
       * @returns {boolean} True if the value is a valid hostname
       */
      var isHostname = function(value) {
        return /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$/.test(value);
      };

      /**
       * Verifies if the informed value is a valid IP (IPV4 or IPV6).
       *
       * @param {string} value - Value to be validated
       * @returns {boolean} True if the value is a valid IP address (IPV4 or IPV6)
       */
      var isIpAddress = function(value) {
        return /((^\s*((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))\s*$)|(^\s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:)))(%.+)?\s*$))/.test(value);
      };
    }
  }
})();


```